### PR TITLE
Switch to controlled prop for lightbox open

### PR DESCRIPTION
**Pull request by pjeweb**
## Describe your changes

...

## Notes for testing your change

...

### CODE REVIEW

File: apps/website/src/components/content/YouTube.tsx [hunk 1]
```diff
@@ -58,34 +57,23 @@ type TriggerProps = {
...
-          {...{ [`data-lightbox-${id}`]: JSON.stringify({ videoId, caption }) }}
-        >
-          {children}
-        </a>
-      );
-    },
-  );
+  const Trigger = ({ videoId, caption, className, children }: TriggerProps) => {
+    // Expose a method to open the lightbox as the ref
+    const elm = useRef<HTMLAnchorElement>(null);
```
Comment (pjeweb) [inline apps/website/src/components/content/YouTube.tsx:62]: forgot to remove this

File: apps/website/src/components/content/YouTube.tsx [hunk 2]
```diff
@@ -143,33 +131,25 @@ type LightboxCtxProps = {
 
 type LightboxProps = {
   id?: string;
-  onInit?: () => void;
-  onDestroy?: () => void;
   className?: string;
   children: ReactNode | ((ctx: LightboxCtxProps) => ReactNode);
+  defaultOpen?: number;
```
Comment (pjeweb) [inline apps/website/src/components/content/YouTube.tsx:136]: maybe there is another name. its not a controlled value so i tried to use the defaultX naming convention like its common for html elements?!

**Comment by MattIPv4** [comment]
Perhaps we update `Trigger` to require a unique key, and then can have the prop on `Lightbox` take that key and handle doing the DOM lookup for the index itself, so the consumer doesn't have to worry about how many `Trigger`s they're using, and in what order they're being rendered to the DOM etc.?

File: apps/website/src/components/content/YouTube.tsx (hunk 2 above)
Comment (MattIPv4) [reply apps/website/src/components/content/YouTube.tsx:130]: I very much am not a fan of non-stateful props like this -- if we wanted to drive this by a prop, I'd probably want an `open` prop as well as an `onChange`, and expect them to always be in sync. If we just want something we use on demand, it should be a method not a prop imo.

Comment (pjeweb) [reply apps/website/src/components/content/YouTube.tsx:130]: sure. that should be quite possible to make it synced/controlled

File: apps/website/src/components/content/YouTube.tsx [hunk 3]
```diff
@@ -244,44 +219,22 @@ export const Lightbox = ({
...
+  // Open the lightbox on load if we have a default
+  useEffect(() => {
+    if (defaultOpen && defaultOpen > -1) {
+      const gallery = document.getElementById(
+        photoswipeId,
+      ) as HTMLElement | null;
+      if (!gallery) return;
+      lightboxRef.current?.loadAndOpen(defaultOpen, { gallery });
+    }
+  }, [defaultOpen, photoswipeId]);
```
Comment (MattIPv4) [inline apps/website/src/components/content/YouTube.tsx:237]: We probably want to wait until the lightbox is initialized before doing this, and only handle if the useEffect for the lightbox gets re-run? The ref is probably fine, but we probably also still want a state to track once its initialized?

**Comment by pjeweb** [comment]
> Perhaps we update `Trigger` to require a unique key, and then can have the prop on `Lightbox` take that key and handle doing the DOM lookup for the index itself, so the consumer doesn't have to worry about how many `Trigger`s they're using, and in what order they're being rendered to the DOM etc.?

Yeah that would probably make it more robust. The trigger could get the container/parent selector in `createTrigger` and all the trigger elements would need some attribute to find them. Makes sense.

File: apps/website/src/components/content/YouTube.tsx (hunk 3 above)
Comment (pjeweb) [reply apps/website/src/components/content/YouTube.tsx:237]: I would've assumed the effects are done in series and they the init is synchronous, so it should be mostly guaranteed to be init'ed at that point?

Comment (MattIPv4) [reply apps/website/src/components/content/YouTube.tsx:237]: I would think so, but given they're different hooks with different dependencies, I would prefer to make this robust?

Comment (pjeweb) [reply apps/website/src/components/content/YouTube.tsx:237]: sure, also i dont think your approach is bad. with this one the arrow buttons in the lightbox does not seem to work. its probably not as easy as i thought ;)

**Comment by MattIPv4** [comment]
@pjeweb pushed a commit here that I think does what we discussed above. Feel free to hack around or w/e, lmk if you're happy to merge this into the other branch

**Comment by pjeweb** [comment]
> @pjeweb pushed a commit here that I think does what we discussed above. Feel free to hack around or w/e, lmk if you're happy to merge this into the other branch

I would be happy to merge this! Looks good and works very well.

**Pull request merged**